app/services/api/webflow-api.ts

```diff
@@ -31,62 +25,108 @@ import {
   cleanedWorkshops,
   convertScheduleToScheduleCard,
 } from "./webflow-helpers"
+import { fromZodError } from "zod-validation-error"
+import type {
+  RecommendationsCollection,
+  RecurringEventsCollection,
+  SpeakersCollection,
+  SpeakerNamesCollection,
+  SponsorsCollection,
+  TalksCollection,
+  VenuesCollection,
+  WorkshopsCollection,
+  ScheduledeventsCollection,
+} from "./webflow-api.generated"
+import { Webflow } from "./webflow-api.service"
 
-const useWebflowAPI = <T>(key: string, collectionId: string, enabled = true) =>
-  useQuery({
+const webflow = Webflow({ api: axiosInstance })
+
+const useWebflowAPI = <T>(key: CollectionKey, collectionId: CollectionId, enabled = true) =>
+  useQuery<T[], ZodError | AxiosError, T[], CollectionKey[]>({
     queryKey: [key],
     queryFn: async () => {
-      const { data } = await axiosInstance.get<PaginatedItems<T>>(
-        `/collections/${collectionId}/items`,
-      )
-      return data.items
+      const schema = COLLECTIONS_MAP[key].schema
+      try {
+        const payload = await webflow.collection.items.get({ collectionId, itemSchema: schema })
+        return payload.items as T[]
+      } catch (error) {
+        const validationError = fromZodError(error)
+        throw Error(
+          `Whoops! Our API didn't come back in the expected shape. ${
+            schema.description ?? "Unknown schema"
+          } threw the following error: ${validationError.message}`,
+        )
+      }
     },
     enabled,
   })
```

Comment:
During the rebase, this function got a little mangled merging the two. This now works.